### ISSUE TITLE
refactor IRC reactor out of mb2 proper

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#41]
Right now the web client and IRC reactor both live in the same process and talk to the heart of mb2 by using strings which are then converted into enums which the heart processes. Logically I want the IRC reactor to be a completely separate process and for it to parse the strings into enums which are then sent through a web-socket to the heart of mb2. Similarly, I want the web client to use UI elements (instead of text) to create enums which are then sent through a web-socket to the heart of mb2.

IOW, the only interface mb2 should provide is a web-socket one, and all clients will speak it. This will allow me to incrementally shift the web-client from text-based to UI based. It will also make it trivial for me to not use non-breaking spaces in the web-client when displaying chat (#38).

So, there won't really be much user visible change from me implementing this issue, but I won't really be doing much with the web interface until I've done this refactor.

**Comment by ctm** [comment]
FWIW, I've done a lot of work on this ticket and am happy with the progress I've made, although so far there is nothing visible from the outside (and the IRC reactor is still baked in). I'm hopeful that this work will bear fruit before Sunday, December 8th's play testing, however my schedule is a bit tweaked due to a combination of my wife heading out of town and me needing to take a CiSAR class on Saturday.

**Comment by ctm** [comment]
It was clear that I wouldn't get this done in time for this week's play testing, so I got to a good stopping point and then did some other work needed to support more tournament formats. I did manage to get that work done in time for this week's play testing, which makes me happy because that allows me to continue adding tournament structures when I have blocks of time to code that aren't big enough for some of my larger projects.

**Comment by ctm** [comment]
I need to get back to this ticket, because it's a blocker for #10

**Comment by ctm** [comment]
This one is taking longer than I expected for a number of reasons, the biggest probably being that I simply underestimated how much (typically mindless) work there was to do. However, I did get surprised by the state of binary encodings in yew and serde_cbor, so I spent a little time sorting that out (with PRs accepted to both).

FWIW, I'm ripping an entire small subsystem (crier) out and making better use of actix-web's features, now that I know more than I did back when I started this project. I'm also making it so that each table gets its own window. This restructure will also make it easy to implement observers (#49 ), which is good, because adding IRC back in after this refactor may take a little time, not because it will be hard to do, but because nobody is using it and the door will be open to a ton of UI improvements after I'm done here.
